packages/catalog-server/src/lib/catalog.ts

```diff
@@ -144,16 +155,43 @@ export class Catalog {
       }
 
       // Write the tags
+      console.log('Writing package dist tags...');
       await this.#repository.updateDistTags(
         packageName,
         [...versionsToUpdate],
         newDistTags
       );
+      console.log('  done');
     }
 
+    let importResult:
+      | {packageVersion?: PackageVersion; problems?: ValidationProblem[]}
+      | undefined = undefined;
+
     if (versionToImport !== undefined) {
-      await this.importPackageVersion(packageName, versionToImport);
+      importResult = await this.importPackageVersion(
+        packageName,
+        versionToImport
+      );
     }
+
+    console.log('Marking package ready...');
+    const newPackageInfo: ReadablePackageInfo = {
```

Comment:
Just to clarify -- these changes weren't in the PR description -- we weren't setting the ready state before?

Reply:
Yeah... need to add tests!